src/Lookup/ClosestReferencedEntityIdLookup.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * Service interface for getting the closest entity (out of a specified set),
+ * from a given starting entity. The starting entity, and the target entities
+ * are (potentially indirectly, via intermediate entities) linked by statements
+ * with a given property id.
+ *
+ * @since 3.10
+ *
+ * @license GPL-2.0+
+ * @author Marius Hoch
+ */
+interface ClosestReferencedEntityIdLookup {
+
+	/**
+	 * Get the closest entity (out of $toIds), from a given entity. The starting entity, and
+	 * the target entities are (potentially indirectly, via intermediate entities) linked by
+	 * statements with the given property id.
+	 *
+	 * @since 3.10
+	 *
+	 * @param EntityId $fromId
+	 * @param EntityId[] $toIds
+	 * @param PropertyId $propertyId
+	 *
+	 * @return EntityId|null
```

Comment:
How will this distinguished between “no target entity found” and “search limit exhausted”? (I assume the limit itself will be implementation-specific, but the interface should still provide an appropriate return type, right?)

Reply:
I thought it's probably best to throw an exception in that case?! That would make the interface agnostic from that.

Reply:
Okay, an exception makes sense. And I agree that the limit itself doesn’t belong in the interface.